example/particles/particles.hpp

```diff
@@ -28,11 +28,18 @@ namespace particles_example {
 
 typedef Kokkos::Random_XorShift64_Pool<> RNGPool;
 
-class ParticleDriver : public MultiStageDriver {
+class ParticleDriver : public EvolutionDriver {
  public:
   ParticleDriver(ParameterInput *pin, ApplicationInput *app_in, Mesh *pm)
-      : MultiStageDriver(pin, app_in, pm) {}
-  TaskCollection MakeTaskCollection(BlockList_t &blocks, int stage);
+      : EvolutionDriver(pin, app_in, pm),
+        integrator(std::make_unique<StagedIntegrator>(pin)) {}
+  TaskCollection MakeParticlesCreationTaskCollection();
+  TaskCollection MakeParticlesUpdateTaskCollection();
+  TaskCollection MakeFinalizationTaskCollection();
+  TaskListStatus Step();
```

Comment:
I didn't find this too painful, though maybe this smells off in a way I don't recognize. I don't think this can be reduced at the moment, because we need to have tasks on either side of `MakeParticlesUpdateTaskCollection`, which we loop over in `Step()`. I suspect this will change as we think more about how to build task lists for iterative methods -- this example may be a good place to experiment with that.

Reply:
👍